File: erlang/numbers/src/numbers_recursive.erl
-module(numbers_recursive).
-export([fibonacci/1, factorial/1, sum_numbers/1]).

fibonacci(N) when N =< 1 ->
    N;
fibonacci(N) ->
    fibonacci(N - 1) + fibonacci(N - 2).

factorial(N) when N =< 1 ->
    1;
factorial(N) ->
    N * factorial(N - 1).

sum_numbers(N) when N =< 0 ->
    0;
sum_numbers(N) ->
    N + sum_numbers(N - 1).

File: haskell/numbers/src/NumbersRecursive.hs
module NumbersRecursive (fibonacci, factorial, sumNumbers) where

fibonacci :: Int -> Int
fibonacci 0 = 0
fibonacci 1 = 1
fibonacci n = fibonacci (n - 1) + fibonacci (n - 2)

factorial :: Int -> Int
factorial 0 = 1
factorial n = n * factorial (n - 1)

sumNumbers :: Int -> Int
sumNumbers 0 = 0
sumNumbers n = n + sumNumbers (n - 1)